store/src/store.rs

```diff
@@ -236,7 +250,19 @@ impl<DB: Database> KVStore<DB> {
     }
 }
 
-pub(crate) enum AnyKVStore<'a, DB: Database> {
+/// Equivalent to [`BasicKVStore`](https://docs.cosmos.network/v0.46/core/store.html#base-layer-kvstores) from cosmos
+pub trait KVStoreTrait {
+    fn get(&self, k: &impl AsRef<[u8]>) -> Option<Vec<u8>>;
+    // fn contains( &self, k : &impl AsRef<[u8]>) -> bool;
+    fn set(&mut self, key: impl IntoIterator<Item = u8>, value: impl IntoIterator<Item = u8>);
+    // fn delete( &mut self, k : &impl AsRef<[u8]> );
+}
+
+pub trait AnyStoreReadTrait<DB: Database, SK> {
```

Comment:
I think we should get rid of this trait. It solves the same problem as the `KVStoreTrait` but in a way that's difficult to follow.

Reply:
Removed